logictest: statement_statistics failure with `local-spec-planning`

Running `statement_statistics` with a config other than `local` should work. It seems that in some configs, `SELECT x FROM test WHERE y IN (_, _, __more3__)` is dropped. This is the case with the `local-spec-planning` and `3node-tenant`.

```
    TestLogic/local-spec-planning/statement_statistics: logic.go:2063:

        testdata/logic_test/statement_statistics:109: SELECT key,flags
          FROM test.crdb_internal.node_statement_statistics
         WHERE application_name = 'valuetest' ORDER BY key, flags
        expected:
            key                                                               flags
            INSERT INTO test VALUES (_, _, __more1__), (__more1__)            ·
            SELECT (_, _, __more3__) FROM test WHERE _                        ·
            SELECT key FROM test.crdb_internal.node_statement_statistics      ·
            SELECT sin(_)                                                     ·
            SELECT sqrt(_)                                                    !
            SELECT x FROM (VALUES (_, _, __more1__), (__more1__)) AS t (x)    ·
            SELECT x FROM test WHERE y = (_ / z)                              !+
            SELECT x FROM test WHERE y IN (_, _, _ + x, _, _)                 ·
            SELECT x FROM test WHERE y IN (_, _, __more3__)                   ·
            SELECT x FROM test WHERE y IN (_, _, __more3__)                   +
            SELECT x FROM test WHERE y NOT IN (_, _, __more3__)               ·
            SET CLUSTER SETTING "debug.panic_on_failed_assertions" = DEFAULT  ·
            SET CLUSTER SETTING "debug.panic_on_failed_assertions" = _        ·
            SET application_name = _                                          ·
            SET distsql = "on"                                                ·
            SHOW CLUSTER SETTING "debug.panic_on_failed_assertions"           ·
            SHOW application_name                                             ·

        but found (query options: "colnames") :
            key                                                               flags
            INSERT INTO test VALUES (_, _, __more1__), (__more1__)            ·
            SELECT (_, _, __more3__) FROM test WHERE _                        ·
            SELECT key FROM test.crdb_internal.node_statement_statistics      ·
            SELECT sin(_)                                                     ·
            SELECT sqrt(_)                                                    !
            SELECT x FROM (VALUES (_, _, __more1__), (__more1__)) AS t (x)    ·
            SELECT x FROM test WHERE y = (_ / z)                              !
            SELECT x FROM test WHERE y IN (_, _, _ + x, _, _)                 ·
            SELECT x FROM test WHERE y IN (_, _, __more3__)                   ·
            SELECT x FROM test WHERE y NOT IN (_, _, __more3__)               ·
            SET CLUSTER SETTING "debug.panic_on_failed_assertions" = DEFAULT  ·
            SET CLUSTER SETTING "debug.panic_on_failed_assertions" = _        ·
            SET application_name = _                                          ·
            SET distsql = "on"                                                ·
            SHOW CLUSTER SETTING "debug.panic_on_failed_assertions"           ·
            SHOW application_name                                             ·

--- done: testdata/logic_test/statement_statistics with config local-spec-planning: 26 tests, 1 failures

    TestLogic/local-spec-planning/statement_statistics: logic.go:2644:
        testdata/logic_test/statement_statistics:133: error while processing
    TestLogic/local-spec-planning/statement_statistics: logic.go:2644: testdata/logic_test/statement_statistics:133: too many errors encountered, skipping the rest of the input
```

Jira issue: CRDB-7997

This test still fails in 3node-tenant but now it appears only because of the flags. The `+` flag is for DistSQL which isn't available in tenant mode.

```
        --- FAIL: TestTenantLogic/3node-tenant/statement_statistics (0.76s)
            logic.go:2400: 
                 
                ../../sql/logictest/testdata/logic_test/statement_statistics:113: SELECT key,flags
                  FROM test.crdb_internal.node_statement_statistics
                 WHERE application_name = 'valuetest' ORDER BY key, flags
                expected:
                    key                                                               flags
                    INSERT INTO test VALUES (_, _, __more1__), (__more1__)            ·
                    SELECT (_, _, __more3__) FROM test WHERE _                        ·
                    SELECT key FROM test.crdb_internal.node_statement_statistics      ·
                    SELECT sin(_)                                                     ·
                    SELECT sqrt(_)                                                    !
                    SELECT x FROM (VALUES (_, _, __more1__), (__more1__)) AS t (x)    ·
                    SELECT x FROM test WHERE y = (_ / z)                              !+
                    SELECT x FROM test WHERE y IN (_, _, _ + x, _, _)                 ·
                    SELECT x FROM test WHERE y IN (_, _, __more3__)                   +
                    SELECT x FROM test WHERE y NOT IN (_, _, __more3__)               ·
                    SET CLUSTER SETTING "debug.panic_on_failed_assertions" = DEFAULT  ·
                    SET CLUSTER SETTING "debug.panic_on_failed_assertions" = _        ·
                    SET application_name = '_'                                          ·
                    SET distsql = "on"                                                ·
                    SHOW CLUSTER SETTING "debug.panic_on_failed_assertions"           ·
                    SHOW application_name                                             ·
                    
                but found (query options: "colnames") :
                    key                                                               flags
                    INSERT INTO test VALUES (_, _, __more1__), (__more1__)            ·
                    SELECT (_, _, __more3__) FROM test WHERE _                        ·
                    SELECT key FROM test.crdb_internal.node_statement_statistics      ·
                    SELECT sin(_)                                                     ·
                    SELECT sqrt(_)                                                    !
                    SELECT x FROM (VALUES (_, _, __more1__), (__more1__)) AS t (x)    ·
                    SELECT x FROM test WHERE y = (_ / z)                              !
                    SELECT x FROM test WHERE y IN (_, _, _ + x, _, _)                 ·
                    SELECT x FROM test WHERE y IN (_, _, __more3__)                   ·
                    SELECT x FROM test WHERE y NOT IN (_, _, __more3__)               ·
                    SET CLUSTER SETTING "debug.panic_on_failed_assertions" = DEFAULT  ·
                    SET CLUSTER SETTING "debug.panic_on_failed_assertions" = _        ·
                    SET application_name = '_'                                        ·
                    SET distsql = "on"                                                ·
                    SHOW CLUSTER SETTING "debug.panic_on_failed_assertions"           ·
                    SHOW application_name                                             ·
            logic.go:3282: 
                ../../sql/logictest/testdata/logic_test/statement_statistics:136: error while processing
```

--- a/pkg/sql/logictest/testdata/logic_test/statement_statistics
+++ b/pkg/sql/logictest/testdata/logic_test/statement_statistics
@@ -151,7 +151,6 @@ SET application_name = '';
 statement ok
 RESET distsql
 
-skipif config 3node-tenant-default-configs #52763
 query TT colnames
 SELECT key,flags
   FROM test.crdb_internal.node_statement_statistics

